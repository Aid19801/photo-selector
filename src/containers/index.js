import HomePage from './homepage/homepage';
import PhotoGallery from './photo-gallery/photo-gallery';

export {
    HomePage,
    PhotoGallery
}