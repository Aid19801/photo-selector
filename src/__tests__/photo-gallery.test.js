import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store'

import { PhotoGallery } from '../containers';

describe('Photo Gallery', () => {

    // photoPage's portion of the Store.
    let initialState = { photosPage: {
        isLoadingPage: false,
        isLoadingPhotos: false,
        isError: false,
        photos: [{ title: 'mock-title', url: 'http://mock.net', thumbnailUrl: 'http://thumbnail.net' }]
    }};
    
    const mockStore = configureStore()
    let store;
    let container;

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<PhotoGallery store={store} /> )  
    });

    it('should render the connected(PhotoGallery) component without exploding', () => {
        expect(container.length).toEqual(1)
        expect(container).toBeTruthy();
    });

    it('should render the expected props', () => {
        const expectedProp = [{ title: 'mock-title', url: 'http://mock.net', thumbnailUrl: 'http://thumbnail.net' }]
        
        expect(container.props().isLoadingPage).toBe(false);
        expect(container.props().isLoadingPhotos).toBe(false);
        expect(container.props().isError).toBe(false);
        expect(container.props().photos).toEqual(expectedProp);
    });

    it('should render the expected child elements', () => {
        let mountedContainer = mount(<PhotoGallery store={store} /> )  
        expect(mountedContainer.find('div').length).toBe(6);
        expect(mountedContainer.find('img').length).toEqual(1);
        expect(mountedContainer.find('p').length).toEqual(1);
    });

    it('should match the quick n dirty snapshot...', () => {
        let snapShot = renderer.create(container).toJSON();
        expect(snapShot).toMatchSnapshot();
    });
});
