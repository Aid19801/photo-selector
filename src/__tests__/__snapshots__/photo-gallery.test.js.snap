// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo Gallery should give user option to select a photo 1`] = `
<div
  className="galleryContainer"
>
  <div
    className=""
    onScroll={[Function]}
    style={Object {}}
  >
    <div
      style={Object {}}
    >
      <div
        style={
          Object {
            "height": 0,
            "width": "100%",
          }
        }
      />
      <div
        className="eachPhotoContainer"
        onClick={[Function]}
      >
        <img
          className="img"
          src="http://thumbnail.net"
        />
        <p
          className="title"
        >
          mock-title
        </p>
      </div>
      <div
        style={
          Object {
            "height": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Photo Gallery should match the quick n dirty snapshot... 1`] = `
<div
  className="galleryContainer"
>
  <div
    className=""
    onScroll={[Function]}
    style={Object {}}
  >
    <div
      style={Object {}}
    >
      <div
        style={
          Object {
            "height": 0,
            "width": "100%",
          }
        }
      />
      <div
        className="eachPhotoContainer"
        onClick={[Function]}
      >
        <img
          className="img"
          src="http://thumbnail.net"
        />
        <p
          className="title"
        >
          mock-title
        </p>
      </div>
      <div
        style={
          Object {
            "height": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
